### ISSUE TITLE
next.js build: building contentlayer prior to TSC or adding more clear docs

### ISSUE THREAD

**Issue by jzxhuang** [contentlayerdev/contentlayer#494]
When following the official Getting Started example for Next.js, the app fails to build on CI or any environemnt where `.contentlayer` doesn't exist already.

This appears to be because TSC runs _prior_ to contentlayer, which errors out. 

I'm not sure if this can be changed at all (to run a Next plugin prior to the TSC step). That would seem like the ideal solution, but otherwise, would be great to update the documentation & examples to modify the build step 

```json
"scripts": {
  ...
  "build": "contentlayer && next build"
}
```

I'm happy to make a PR to help out, just wanted to check if this is indeed an issue that others experience or if I'm really missing something basic

**Comment by stackbit-admin** [comment]
@jzxhuang Where would you expect to see this documentation? 

Perhaps a mention in the getting started guide and also a note in _Known Problems_ or maybe a separate troubleshooting page?

**Comment by seancdavis** [comment]
@jzxhuang Where would you expect to see this documentation? 

Perhaps a mention in the getting started guide and also a note in _Known Problems_ or maybe a separate troubleshooting page?

**Comment by seancdavis** [comment]
@shaneafsar Can you say more about your development setup?

- What errors were you running into? 
- How did concurrently fix those errors?
- Why did you choose concurrently over other options for this type of behavior?

**Comment by jzxhuang** [comment]
> @jzxhuang Where would you expect to see this documentation?
> 
> Perhaps a mention in the getting started guide and also a note in _Known Problems_ or maybe a separate troubleshooting page?

Yeah, I would expect it in the `Getting Started` guide — probably under "next.js configuration" or a new section like "Building for Production" or something like that. Also adding a note in `Known Problems` is a great idea!

btw, happy to make a PR for this — I just wanted to first confirm that it's a known issue with any easy solutions, and not just something I'm missing 😅
